src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,176 @@
-import React, { useState, useCallback, useRef } from "react";
+import React, { useCallback, useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { Banner } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+
+import { ConfirmationBannerModal } from "./ConfirmationBannerModal";
+
+import "./BannerAdmin.scss";
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const inputFieldRef = useRef<HTMLInputElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const { register, handleSubmit, errors, reset, watch } = useForm<Banner>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isActionButtonValue = watch("isActionButton");
+
+  const [bannerData, setBannerDate] = useState<Banner>();
+
+  const {
+    isShown: isShowModal,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
+  const [isDisabledUrlFields, setIsDisabledUrlFields] = useState(
+    venue?.banner?.isActionButton
+  );
 
-  const handleInputChange = useCallback(() => {
-    setError(null);
-  }, []);
+  useEffect(() => {
+    setIsDisabledUrlFields(isActionButtonValue);
+  }, [isActionButtonValue]);
```

Comment:
What is this `useState`/`useEffect` needed for?

Reply:
basically it enables/disables inputs that belong to button url/title if that option changes value

Reply:
It’s seems unneeded. Just check the variable itself directly? Why add extra useEffect + useState

Reply:
@0xdevalias initial value of `isActionButtonValue` on render is always false since react hook form's `watch `happens when form state changes, thus it breakes ui logic
with this state & default value of `venue?.banner?.isActionButton` we can ensure that on first load  and on any change that status is correct

Reply:
yep, thank you
I've added default value to the `useForm`'s watch, so that variable in question covers this use case 1c2d835